developer-guide.md

```diff
@@ -0,0 +1,545 @@
+    - [Validate Connection](#validate-connection)
+- [Indexing & Querying](#indexing--querying)
+  - [Create an Index](#create-an-index)
+  - [Inserting Documents](#inserting-documents)
+  - [Querying Documents](#querying-documents)
+- [Data Modeling](#data-modeling)
+  - [Long unstructured text](#long-unstructured-text)
+- [Tweaking Relevance](#tweaking-relevance)
+  - [Semantic Search](#semantic-search)
+  - [Text Search](#text-search)
+  - [Hybrid Search](#hybrid-search)
+  - [Search with Sparse Vector and ELSER](#search-with-sparse-vector-and-elser)
+  - [Boosting & Re-ranking](#boosting--re-ranking)
+- [Filtering](#filtering)
+
+# Integrations
+
+- [Integrations](#integrations)
+  - [OpenAI](#openai)
+  - [Hugging Face](#hugging-face)
+  - [Langchain](#langchain)
+
+## Introduction
+
+This guide is intended for developers who want to use the Gen AI with Elasticsearch API to build applications that use the Gen AI with Elasticsearch service.
+
+Elasticsearch is a search database that allows you to store and retrieve relevant information extremely quickly. It is used by many of the world's largest companies to power their search and analytics applications.
+
+Elasticsearch supports many different types of queries, including vector search, full-text search, geospatial search, and more.
+
+### Can Elasticsearch support Vector Search?
+
+Yes! Elasticsearch supports vector search through the [dense_vector](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html) data type.
+
+### Can I do Semantic Search with Elasticsearch?
+
+Yes! With using an embedding model like [sentence-transformers](https://www.sbert.net/) you can transform your text into a vector and store it in Elasticsearch. Then you can use Elasticsearch's vector search capabilities to find similar documents.
+
+### Can I combine both Semantic Search and Full-Text Search?
+
+Yes! You can combine both Semantic Search and Full-Text Search via a "hybrid" query strategy. You retrieve relevant documents through both a vector search and a full-text search, and then balance the scores from both queries to get the final ranking of documents.
+
+### Get started
+
+To get started, [go to the quickstart guide](#quick-start) to setup Elasticsearch and start using the API Client.
+
+## Quick Start
+
+This guide explains how to index and search
+
+**NOTE** This guide assumes you have got [Elasticsearch setup](#setup), you can start using the API Client to index and search documents.
+
+You can quick start in two ways:
+
+- Use the [Quick Start notebook](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/00-quick-start.ipynb) to index and perform Semantic search with Elasticsearch.
+- Follow the commands below in your local development environment.
+
+### Create an Index
+
+An index is a collection of documents that have similar characteristics. For example, you might have an index for "products" and another index for "users".
```

Comment:
```suggestion
Elasticsearch stores content in JSON documents, made up of a number of fields, in an **Elasticsearch index**. An index is similar to a table in relational databases. Use an Elasticsearch index to store documents that have similar characteristics. For example, you might have an index for "products" and another index for "users".
Every index has a _mapping_ which defines the data types and properties of the fields in a document, serving as a schema that dictates how data is indexed and stored.
```

Reply:
Just some optional explanations

Reply:
its not so much JSON documents. Would omit this.